CitiesHarmony.API/SubscriptionPrompt.cs

```diff
@@ -102,15 +127,28 @@ public static class SubscriptionPrompt {
             UIView.library.ShowModal<ExceptionPanel>("ExceptionPanel").SetMessage("Missing dependency: Harmony", message, false);
         }
 
-        private static bool RequiresHarmony2(Assembly assembly) {
+        private static bool RequiresHarmony2(Assembly assembly)
+        {
             if (assembly.GetName().Name == "0Harmony" || assembly.GetName().Name == "CitiesHarmony.Harmony") return false;
 
-            foreach (var assemblyName in assembly.GetReferencedAssemblies()) {
-                if ((assemblyName.Name == "0Harmony" || assemblyName.Name == "CitiesHarmony.Harmony") && assemblyName.Version.Major >= 2) {
+            foreach (var assemblyName in assembly.GetReferencedAssemblies())
+            {
+                if ((assemblyName.Name == "0Harmony" || assemblyName.Name == "CitiesHarmony.Harmony") && assemblyName.Version.Major >= 2)
+                {
                     return true;
                 }
             }
             return false;
         }
+        private static string GetModName(Assembly assembly)
+        {
+            foreach(var plugin in PluginManager.instance.GetPluginsInfo())
```

Comment:
Can you wrap this foreach in a try-catch?

I remember occasions where PluginManager/userModInstance threw exceptions.